### PR TITLE
fix: make logging optional if no logtype passed in

### PR DESCRIPTION
**Pull request by danielgent**
The middleware solution is very noisy and doesn't match what Auth0 does

### CODE REVIEW

File: src/routes/tsoa/authenticate.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class AuthenticateController extends Controller {
    * @returns
    */
   @Post("authenticate")
-  @Middlewares(loggerMiddleware(LogTypes.SUCCESS_CROSS_ORIGIN_AUTHENTICATION))
```
Comment (danielgent) [inline src/routes/tsoa/authenticate.ts:59]: We don't actually want logging here.  It's not a successful cross-origin authentication. That only happens on the `/authorize/ route when the ticket is used

The only logging we want on this route is if an incorrect code is used

File: src/tsoa-middlewares/logger.ts [hunk 2]
```diff
@@ -220,6 +220,8 @@ export function loggerMiddleware(
       const response = await next();
 
       const logType = ctx.var.logType || logTypeInitial;
+      // ir no logtype set then do not log
+      if (!logType) return response;
```
Comment (danielgent) [inline src/tsoa-middlewares/logger.ts:224]: We don't actually want everything logging. I think this middleware solution is much more effort than just simply making calls right before returning or when erroring.  

All the optional vars and context make this much more complex. I think we could just add the logs inline in the data adapter using strict types

It's done now I suppose

Comment (markusahlstrand) [reply src/tsoa-middlewares/logger.ts:224]: I'm thinking if we should migrate away from tsoa at some point and then this logging approach won't really work anyway...

**Review by markusahlstrand — APPROVED** [review]
Looks good

**Pull request merged**